src/core/database/query/match-project-based-props.ts

```diff
@@ -187,3 +187,23 @@ export const rankSens = (variable: string) => oneLine`
     when 'Low' then 1
   end
 `;
+
+export const countEngagements = (nodeName: string) => {
+  return (query: Query) =>
+    query.subQuery(nodeName, (sub) =>
+      sub
+        .match([
+          node(nodeName),
+          relation('out', '', 'engagement'),
+          node('engagement', 'LanguageEngagement'),
+        ])
+        .return('count(engagement) as engagementTotal'),
+    );
+};
```

Comment:
I would probably just inline this in the sorting spot. I don't see too many use cases for this, and it's not abstracting that much.
```ts
engagements: (query) =>
  query
    .match([
      node('node'),
      relation('out', '', 'engagement'),
      node('engagement', 'LanguageEngagement'),
    ])
    .return<{ sortValue: number }>('count(engagement) as sortValue'),
```